p-interface/src/lib.rs

```diff
@@ -8,3 +8,35 @@ pub mod state;
 pub mod program {
     pinocchio_pubkey::declare_id!("TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA");
 }
+
+/// A "dummy" function with a hint to the compiler that it is unlikely to be
+/// called.
+///
+/// This function is used as a hint to the compiler to optimize other code paths
+/// instead of the one where the function is used.
+#[cold]
+pub const fn unlikely_branch() {}
```

Comment:
cosmetic: should we rename this to `cold_path` to match the name used in the experimental API in [core::hint](https://doc.rust-lang.org/core/hint/index.html)?